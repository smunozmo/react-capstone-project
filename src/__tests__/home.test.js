import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../configureStore';
import '@testing-library/jest-dom/extend-expect';
import Home from '../pages/home';

describe('React Capstone Project', () => {
  it('Renders title in Home page:', () => {
    render(<Provider store={store}><BrowserRouter><Home /></BrowserRouter></Provider>);
    expect(screen.getByText('Rick and Morty Characters App')).toBeInTheDocument();
  });

  it('Home snapshot test', () => {
    const myrender = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    expect(myrender).toMatchSnapshot();
  });
});
