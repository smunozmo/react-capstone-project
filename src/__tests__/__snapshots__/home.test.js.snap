// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Capstone Project Home snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <p
          class="text-center p-2"
        >
          Rick and Morty Characters App
        </p>
        <div
          class="row py-5"
        >
          <div
            class="col-6"
          />
          <div
            class="col-6"
          >
            <p
              class="mainStat"
            >
              <span
                class="bg-info p-1"
              >
                Total characters:  
              </span>
            </p>
            <div
              class="spinner-border"
              role="status"
            >
              <span
                class="visually-hidden"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <p>
          <span
            class="shadow bg-info p-1"
          >
            stats by species
          </span>
        </p>
        <div
          class="text-center fs-3 bg-info mt-5 p-3"
        >
          Fetching data from outer space . . .
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <p
        class="text-center p-2"
      >
        Rick and Morty Characters App
      </p>
      <div
        class="row py-5"
      >
        <div
          class="col-6"
        />
        <div
          class="col-6"
        >
          <p
            class="mainStat"
          >
            <span
              class="bg-info p-1"
            >
              Total characters:  
            </span>
          </p>
          <div
            class="spinner-border"
            role="status"
          >
            <span
              class="visually-hidden"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
      <p>
        <span
          class="shadow bg-info p-1"
        >
          stats by species
        </span>
      </p>
      <div
        class="text-center fs-3 bg-info mt-5 p-3"
      >
        Fetching data from outer space . . .
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
